src/bokeh/models/annotations/geometry.py

```diff
@@ -201,17 +200,23 @@ class PolyAnnotation(Annotation):
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
-    xs = Seq(Float, default=[], help="""
+    xs = Seq(FloatOrDatetime, default=[], help="""
     The x-coordinates of the region to draw.
+
+    Datetime values are also accepted, but note that they are immediately
+    converted to milliseconds-since-epoch.
     """)
 
     xs_units = Enum(CoordinateUnits, default='data', help="""
     The unit type for the ``xs`` attribute. Interpreted as |data units| by
     default.
     """)
 
-    ys = Seq(Float, default=[], help="""
+    ys = Seq(FloatOrDatetime, default=[], help="""
     The y-coordinates of the region to draw.
+
+    Datetime values are also accepted, but note that they are immediately
+    converted to milliseconds-since-epoch.
```

Comment:
I expect the "immediately converted" part is not true with this implementation (in all cases). That behavior was an artifact of the "accepts" version. But please correct me if I am mistaken @mattpap